### ISSUE TITLE
fromTokenizer error: Offset hasn't been loaded yet.

### ISSUE THREAD

**Issue by jessrosenfield** [sindresorhus/file-type#379]
I'm not sure if this is an error [here](https://github.com/sindresorhus/file-type/blob/84c26b373863a774dd5fc3dbbb18ff3c2725e25a/core.js#L361) or in [@tokenizer/range](https://github.com/Borewit/tokenizer-range) but I encountered this transient error reading [this file](https://s3.amazonaws.com/docs.coprocure.us/pipeline/files/omnia/omnia_r190202_frey-scientific/Master%20Agreement_R140101_2019_06_30%2FMaster%20Agreement%20Documents%2FRFP%2014-01%20Response_Pricing.xlsx)

- Node v10.19.0
- "@tokenizer/http": "^0.5.2",
- "file-type": "^14.6.2",

```
Error: Offset 23566261 hasn't been loaded yet.
    at ChunkedFileData.readToBuffer (/var/task/node_modules/@tokenizer/range/lib/chunked-file-data.js:80:15)
    at ChunkedFileData.readToBuffer (/var/task/node_modules/@tokenizer/range/lib/chunked-file-data.js:75:33)
    at /var/task/node_modules/@tokenizer/range/lib/range-request-tokenizer.js:69:28
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async _fromTokenizer (/var/task/node_modules/file-type/core.js:361:7)
```

snippet:
```
module.exports = async function getContentFileType(url) {
  const httpTokenizer = await makeTokenizer(url);
  return await FileType.fromTokenizer(httpTokenizer);
}
```

I got this error ~20 times when processing somewhere on the order of 1000s of files. This wasn't limited to XLSX because most of these errors were PDFs. They all were for the same offset: 23566261.

**Comment by jessrosenfield** [comment]
Ignorant question... does this have anything to do with the removal of defining `avoidHeaderRequests: true` in the readme snippet? https://github.com/sindresorhus/file-type/pull/288/files

If not, can you enlighten me on the reason for this change?

**Comment by Borewit** [comment]
> Ignorant question... does this have anything to do with the removal of defining `avoidHeaderRequests: true` in the readme snippet? 

This is related to Borewit/tokenizer-http#357, which is my ToDo list entry I should document that options.

**Comment by Borewit** [comment]
> I got this error ~20 times when processing somewhere on the order of 1000s of files. This wasn't limited to XLSX because most of these errors were PDFs. They all were for the same offset: 23566261.

Good to hear you all using this fancy advanced functions.
Interesting it is the same offset. Is the error always occurring on this file is parsed via [tokenizer-http](https://github.com/Borewit/tokenizer-http).

**Comment by jessrosenfield** [comment]
Yes, in all of these I am using `makeTokenizer` from tokenizer-http. They all have the same trace.

**Comment by Borewit** [comment]
Code snippet I used to reproduce the issue:
```js
const S3 = require('aws-sdk/clients/s3');
const { makeTokenizer } = require('@tokenizer/s3');
const FileType = require('file-type');

const s3 = new S3();

(async function () {

  // Uploaded this file to my private bucket:
  // https://s3.amazonaws.com/docs.coprocure.us/pipeline/files/omnia/omnia_r190202_frey-scientific/Master%20Agreement_R140101_2019_06_30%2FMaster%20Agreement%20Documents%2FRFP%2014-01%20Response_Pricing.xlsx

  const s3Tokenizer = await makeTokenizer(
    s3,
    {
      Bucket: 'music-metadata',
      Key: 'file-type/Master Agreement_R140101_2019_06_30_Master Agreement Documents_RFP 14-01 Response_Pricing.xlsx',
    },
    {
      disableChunked: false,
    },
  );

  try {
    const fileType = await FileType.fromTokenizer(s3Tokenizer);
    console.log(fileType);  } catch (e) {
    console.error(e);
  }
})();
```

**Comment by Borewit** [comment]
Issue should be solved in [tokenizer-range v0.3.2](https://github.com/Borewit/tokenizer-range/releases/tag/v0.3.2).

@jessrosenfield, please have a look if the code snippet, using [@tokenizer/s3](https://www.npmjs.com/package/@tokenizer/s3) is interesting option for you, instead of [@tokenizer/range ](https://github.com/Borewit/tokenizer-range). Sorry for keeping you waiting so long.
